### PR TITLE
C#: Use custom project setting for C# project files name

### LINKED ISSUE #27030: Changing project name breaks all C# scripts
**Godot version:**
Godot 3.1

**OS/device including version:**
Windows 10

**Issue description:**
Changing Project Settings > Application > Config > Name and running the project gives the error "Cannot instance script because the assembly project is not loaded" to all C# scripts. If you close and open Godot again after this, however, it fixes the issue.

### PR DESCRIPTION
**Pull request by raulsntos**
Backports 9c698629499a9a8fffb74a3ec4c2bfac6b7a1484 to 3.x
Fixes #27030

An important difference from the `master` version is I renamed the settings to be more consistent with how we name C# settings in 3.x:

- `dotnet/project/assembly_name` → `mono/project/assembly_name`
- `dotnet/project/solution_directory` → `mono/project/solution_directory`
- `dotnet/project/c#_project_directory` → `mono/project/c#_project_directory`

Created as a draft since this is not in master yet (pending the merge of the `dotnet6` branch).

### CODE REVIEW

**Comment by neikeq** [comment]
> Created as a draft since this is not in master yet (pending the merge of the `dotnet6` branch).

This doesn't need to wait for that one to be merged.

**Review by neikeq — APPROVED** [review]

**Comment by raulsntos** [comment]
Fair enough, but we usually want changes to be merged in master first and I wasn't sure if that code may change before it gets merged.

**Comment by akien-mga** [comment]
Cherry-picked for 3.5.1.

**Pull request merged**
